modules/renter/contractor/contracts.go

```diff
@@ -331,185 +533,94 @@ func (c *Contractor) threadedContractMaintenance() {
 		return
 	}
 
-	// Figure out which contracts need to be renewed, and while we have the
-	// lock, figure out the end height for the new contracts and also the amount
-	// to spend on each contract.
+	// Create the renewSet and refreshSet, each a list of contracts that need to
+	// be renewed, paired with the amount of money to use in each renewal.
 	//
-	// refreshSet is used to mark contracts that need to be refreshed because
-	// they have run out of money. The refreshSet indicates how much currency
-	// was used previously in the contract line, and is used to figure out how
-	// much additional money to add in the refreshed contract.
+	// The renewSet is specifically contracts which are being renewed because
+	// they are about to expire. And the refreshSet is contracts that are being
+	// renewed because they are out of money.
 	//
-	// The actions inside this RLock are complex enough to merit wrapping them
-	// in a function where we can defer the unlock.
-	type renewal struct {
-		id     types.FileContractID
-		amount types.Currency
-	}
-	var endHeight types.BlockHeight
-	var fundsAvailable types.Currency
-	var renewSet []renewal
-
+	// The contractor will prioritize contracts in the renewSet over contracts
+	// in the refreshSet. If the wallet does not have enough money, or if the
+	// allowance does not have enough money, the contractor will prefer to save
+	// data in the long term rather than renew a contract.
+	var renewSet []fileContractRenewal
+	var refreshSet []fileContractRenewal
+
+	// The rest of this function needs to know a few of the stateful variables
+	// from the contractor, build those up under a lock so that the rest of the
+	// function can execute without lock contention.
 	c.mu.RLock()
 	currentPeriod := c.currentPeriod
```

Comment:
shouldn't need to define `currentPeriod` if we use `contractEndHeight()`

Reply:
" without lock contention"